Lib/fontTools/feaLib/parser.py

```diff
@@ -1283,6 +1284,150 @@ def parse_table_OS_2_(self, table):
             elif self.cur_token_ == ";":
                 continue
 
+    def parse_STAT_design_axis(self):
+        axis_tag = self.expect_tag_()
+        if axis_tag not in ('ital', 'opsz', 'slnt', 'wdth', 'wght') and not \
+                axis_tag.isupper():
+            log.warning(
+                f'Unregistered axis tag {axis_tag} should be uppercase.'
+            )
+        axis_order = self.expect_number_()
+        axis_names = self.expect_stat_name_()
+        return axis_tag, axis_order, axis_names
+
+    def parse_STAT_axis_value_(self):
+        assert self.is_cur_keyword_('AxisValue')
+        block = self.ast.NestedBlock(
+            'AxisValue', self.cur_token_, location=self.cur_token_location_
+        )
+        self.expect_symbol_("{")
+        locations = []
+        axis_names = []
+        flags = []
+        while self.next_token_ != "}" or self.cur_comments_:
+            self.advance_lexer_(comments=True)
+            if self.cur_token_type_ is Lexer.COMMENT:
+                block.statements.append(
+                    self.ast.Comment(self.cur_token_,
+                                     location=self.cur_token_location_)
+                )
+            elif self.is_cur_keyword_('location'):
+                loc = self.parse_STAT_location()
+                locations.append(loc)
+            elif self.is_cur_keyword_('name'):
+                platformID, platEncID, langID, string = self.parse_name_()
+                axis_names.append((platformID, platEncID, langID, string))
+            elif self.is_cur_keyword_('flag'):
+                flags = self.expect_stat_flags()
+            else:
+                if self.cur_token_ != ';':
+                    raise FeatureLibError(f'Unexpected token {self.cur_token_} '
+                                          f'in Axis Value',
+                                          self.cur_token_location_)
+        self.expect_symbol_('}')
+        if not axis_names:
+            raise FeatureLibError('Expected "Axis Name"',
+                                  self.cur_token_location_)
+        if not locations:
+            raise FeatureLibError('Expected "Axis location"',
+                                  self.cur_token_location_)
+        return axis_names, locations, flags, block
+
+    def parse_STAT_location(self):
+        values = []
+        loc_tag = self.expect_tag_()
+        if len(loc_tag.strip()) != 4:
+            raise FeatureLibError(f'Axis tag {self.cur_token_} must be 4 '
+                                  f'characters', self.cur_token_location_)
+
+        while self.next_token_ != ';':
+            if self.next_token_type_ in [Lexer.NUMBER, Lexer.FLOAT]:
+                value = self.expect_stat_values_()
```

Comment:
I‘m not sure why do you need `expect_stat_values_()` at all, since you are already checking the token type, you can simplify this to `self.advance_lexer_()` and read the value from `self.cur_token_`.

Reply:
I made it separate originally because we also had the hyphen for ranges and it seemed worth it to move to a separate `expect_`, but you're right, there's no need now.